sklearn/impute/_base.py

```diff
@@ -261,10 +264,10 @@ def _validate_input(self, X, in_fit):
         else:
             dtype = FLOAT_DTYPES
 
-        if not is_scalar_nan(self.missing_values):
-            force_all_finite = True
-        else:
+        if is_pd_na(self.missing_values) or is_scalar_nan(self.missing_values):
```

Comment:
The estimator tag for `SimpleImputer` needs to be updated reflect this:

https://github.com/scikit-learn/scikit-learn/blob/f33fb0af65380e3360cfc9dd7f291ab59e2ef63b/sklearn/impute/_base.py#L116-L117

The estimator tag is used by meta-estimators to see if an inner estimator supports `nan`.
Given tags machinery, there is no simple way to override the `_BaseImputer._more_tags`. One would need to move `_more_tags` down to the children (`SimpleImputer`, `KNNImputer`, etc.) where `SimpleImputer` would define it as:

```python
def _more_tags(self):
    return {"allow-nan": is_pd_na(self.missing_values) or is_scalar_nan(self.missing_values)}
```

Reply:
Thanks for catching. I updated the `_more_tags` function as suggested. But I don't quite understand the reasoning. Why can't we just add a `_more_tags` function to `SimpleImputer` to override the `_BaseImputer`'s implementation? Could you shed more lights on why it doesn't work?

Reply:
Moved `_more_tags` back to base class as per discussion [here](https://github.com/scikit-learn/scikit-learn/pull/21114#discussion_r717849609).